pkg/federate/kubefed/cluster_test.go

```diff
@@ -0,0 +1,252 @@
+package kubefed
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+	fedv1 "sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+)
+
+type added string
+type updated string
+type removed string
+
+type testClusterEventHandler struct {
+	id     string
+	events chan interface{}
+}
+
+const (
+	secretName = "my-secret"
+)
+
+var _ = Describe("Kubefed Cluster", func() {
+	klog.InitFlags(nil)
```

Comment:
It's a nit, but I believe this belongs more on the test_suite.go file.

Reply:
I tried putting it prior to calling `RunSpecs` in `TestFederator` but it didn't work.

Reply:
What do you mean by didn't work? It seems to work for me in coastguard unit tests :?

Reply:
I mean if you pass in logging args like _-logtostderr=true_ and _-v=2_. The process fails with an error like "**flag provided but not defined: -logtostderr**" unless the option flags are registered. So `klog.InitFlags` must be called before `flag.Parse` - it doesn't work if `klog.InitFlags` is in `TestFederator` b/c `flag.Parse` has already been called by that point. `Describe` functions are global/static vars and thus, apparently, invoked prior to `flag.Parse`. I also tried a `BeforeSuite` function but that didn't work either.